src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,176 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Encapsulates all methods that processes the user's input to perform actions accordingly.
+ */
+public class Parser {
+    /**
+     * Parses the input gotten from the main menu.
+     * @param input user input
+     * @param taskList task list managed by the chatbot
+     * @return process code, where 0 means the program should end and 1 otherwise
+     * @throws DukeException if there are malformed inputs
+     */
+    public static int parseInput(String input, TaskList taskList) throws DukeException {
```

Comment:
Can checkout the java.util.regex package that is used for working with regular expressions in Java such as java.util.regex.Pattern to make your code more readable and manageable.